example/lib/main.dart

```diff
@@ -78,10 +82,35 @@ class _MyAppState extends State<MyApp> {
     }
   }
 
+  Future<void> requestLocationPermission() async {
+    if (kIsWeb) {
+      return;
+    }
+    LocationPermission permission = await Geolocator.checkPermission();
+    if (permission == LocationPermission.denied) {
+      permission = await Geolocator.requestPermission();
+      if (permission == LocationPermission.denied) {
+        // Permissions are denied, next time you could try
+        // requesting permissions again (this is also where
+        // Android's shouldShowRequestPermissionRationale
+        // returned true. According to Android guidelines
+        // your App should show an explanatory UI now.
+        return Future.error('Location permissions are denied');
```

Comment:
I think this is not how we would want users to handle a lack of permissions, consider logging errors instead of throwing them and processing with identification?